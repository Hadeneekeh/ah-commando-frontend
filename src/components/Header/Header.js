/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import Button from '@Components/Button';
import AuthStore from '@Lib/AuthStore';
import Icon from '@Components/Icon';
import connectComponent from '@App/lib/connect-component';
import { openModal } from '@Actions/uiActions';
import { updateSearchQuery, getFilteredArticles, updatePageNumber } from '@Actions/searchActions';
import logo from '../../../public/logo.png';
import './Header.scss';

const buttonStyle = {
  width: '100px',
  height: '35px',
};

export class Header extends Component {
  state = {
    search: false,
  }

  handleClick = () => {
    this.setState({
      search: true,
    });
  };

  handleKeyUp = (e) => {
    if (e.keyCode === 13) {
      if (e.target.value.length) {
        this.props.updateSearchQuery(e.target.value);
        // this.props.getArticles(e.target.value);
        this.props.getFilteredArticles(e.target.value);
        this.props.updatePageNumber(1);
        this.props.history.push('/search');
      }
    }
  };

  handleBlur = () => {
    this.setState({
      search: false,
    });
  }

  render() {
    const { search } = this.state;
    const { signIn, signUp } = this.props;
    return (
      <>
        <header className="header-top">
          <img className="logo" src={logo} alt="" />
          <div className="search">
            <div>
              <div>
                {search ? <input type="text" onKeyUp={(e) => { this.handleKeyUp(e); }} datatest="input-search" placeholder="Search..." onBlur={this.handleBlur} /> : ''}
              </div>
              <button
                type="button"
                className="searchButton"
                onClick={this.handleClick}
                datatest="search-icon"
              >
                <Icon name="search" />
              </button>
            </div>
            <span style={search ? { transform: 'scaleX(1)' } : { transform: 'scaleX(0)' }}>{' '}</span>
          </div>
          {AuthStore.getToken()
            ? (
              <div className="action">
                <button className="notification" type="button" style={buttonStyle} onClick={this.handleClick}>
                  <Icon name="notification" />
                </button>
                <Button style={buttonStyle} handleClick={this.handleClick}>Upgrade</Button>
              </div>
            )
            : (
              <div className="action">
                <Button
                  label="sign in"
                  handleClick={() => signIn('signin')}
                  disabled={false}
                  type="button"
                  style={buttonStyle}
                  id="signin"
                  datatest="signin-button"
                />
                <Button
                  datatest="signup-button"
                  label="sign up"
                  handleClick={() => signUp('signup')}
                  disabled={false}
                  type="button"
                  style={buttonStyle}
                  id="signup"
                />
              </div>
            )}
        </header>
        <div className="navigation">
          <ul>
            <Link to="/a">Technology</Link>
            <Link to="/a">Health</Link>
            <Link to="/a">Culture</Link>
            <Link to="/a">Science</Link>
            <Link to="/a">Fashion</Link>
            <Link to="/a">Education</Link>
            <Link to="/a">Lifestyle</Link>
            <Link to="/a">Nature</Link>
          </ul>
        </div>
      </>
    );
  }
}

Header.propTypes = {
  signIn: PropTypes.func.isRequired,
  signUp: PropTypes.func.isRequired,
};

export default connectComponent(withRouter(Header), {
  signIn: () => openModal('signin'),
  signUp: () => openModal('signup'),
  updateSearchQuery,
  getFilteredArticles,
  updatePageNumber,
});
