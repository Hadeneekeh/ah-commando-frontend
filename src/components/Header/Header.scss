
.header-top {
  width: 100%;
  background-color: #FFF;
  padding: 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  & > div {
    flex-direction: column;
  }

  .action {
    display: flex;
    flex-direction: row;
  }
}

.navigation {
  padding: 10px;

  ul {
    list-style-type: none;
    padding: 0 20px;
    display: flex;
    justify-content: space-between;

    li {
      padding: 10px;
    }
  }
}

