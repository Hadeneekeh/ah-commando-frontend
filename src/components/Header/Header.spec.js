/* eslint-disable react/forbid-foreign-prop-types */
import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';
import { findByTestAttribute } from '@Utils/';


const shallowRender = (props) => {
  const component = shallow(<Header {...props} />);
  return component;
};

describe('Header component', () => {
  let wrapper;
  const expectedprops = {
    signIn: jest.fn(),
    signUp: jest.fn(),
    updateSearchQuery: jest.fn(),
    getFilteredArticles: jest.fn(),
    updatePageNumber: jest.fn(),
    getNotifications: jest.fn(),
    getProfile: jest.fn(),
    history: {
      location: {
        pathname: '/',
      },
      push: jest.fn(),
    },
    profile: {
      user: {
        image: '',
      },
    },
    notifications: {
      notifications: [],
    },
  };
  beforeEach(() => {
    wrapper = shallowRender(expectedprops);
  });

  it('should render the header component without crashing', () => {
    const component = findByTestAttribute(wrapper, 'headerComponent');
    expect(component.length).toBe(1);
  });

  it('should render the navigation component without crashing', () => {
    const component = findByTestAttribute(wrapper, 'navigationComponent');
    expect(component.length).toBe(1);
  });

  describe('handle modal pop up for signin and signup', () => {
    beforeEach(() => {
      wrapper = shallow(<Header {...expectedprops} />);
    });

    it('should open signup modal', () => {
      const signUpBtn = wrapper.find('[datatest="signup-button"]');
      signUpBtn.simulate('click');
      expect(expectedprops.signUp).toMatchSnapshot();
    });

    it('should test handleClick function', () => {
      const instance = wrapper.instance();
      const btn = wrapper.find('.searchButton');
      btn.simulate('click');
      instance.handleClick = jest.fn();
      instance.handleClick();
      expect(instance.handleClick).toHaveBeenCalled();
    });

    it('should test handlekeyUp function', () => {
      const instance = wrapper.instance();
      instance.state = {
        search: true,
      };
      expect(instance.state.search).toBe(true);
      const e = {
        keyCode: 13,
        target: {
          value: 'not-empty',
        },
      };
      instance.handleKeyUp = jest.fn();
      instance.handleKeyUp(e);
      expectedprops.getFilteredArticles(e.target.value);
      expectedprops.updatePageNumber(1);
      expectedprops.updateSearchQuery(e.target.value);
      expectedprops.history.push('/search');
      expect(instance.handleKeyUp).toHaveBeenCalled();
    });

    it('should test handleBlur function', () => {
      const instance = wrapper.instance();
      instance.state = {
        search: true,
      };
      expect(instance.state.search).toBe(true);
      instance.handleBlur = jest.fn();
      instance.handleBlur();
      expect(instance.handleBlur).toHaveBeenCalled();
    });

    it('should test the open search function', () => {
      const instance = wrapper.instance();
      const searchButton = wrapper.find('[datatest="search-icon"]');
      searchButton.simulate('click');
      expect(instance.state.search).toBe(true);
    });
  });
});
