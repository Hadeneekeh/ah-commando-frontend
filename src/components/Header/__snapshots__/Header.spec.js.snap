// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle modal pop up for signin and signup should open signup modal 1`] = `[MockFunction]`;
